Fotoblog.BLL/Services/PhotosService/PhotoService.cs

```diff
@@ -0,0 +1,185 @@
+﻿using AutoMapper;
+using Fotoblog.BLL.Services.ServiceResultNS;
+using Fotoblog.DAL;
+using Fotoblog.DAL.Entities;
+using Fotoblog.Utils.ViewModels.Photos;
+using Microsoft.AspNetCore.Http;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.Processing;
+using System.IO;
+using System.Net;
+
+namespace Fotoblog.BLL.Services.PhotosService
+{
+    public class PhotoService : BaseService, IPhotoService
+    {
+        private readonly string[] _allowedExtensions = { ".jpg", "jpeg", ".png" };
+        private readonly string[] _allowedImgContentTypes = { "image/jpeg", "image/png" };
+        private string? _uploadedFileExtension;
+        private string _saveLocation = "c:\\uploads"; // random folder added later
+        private readonly int _thumbnailWidth = 600;
+        public PhotoService(ApplicationDbContext dbContext, IMapper mapper) : base(dbContext, mapper)
+        {
+        }
+
+        public async Task<ServiceResult> AddPhoto(NewPhotoDataVm newPhotoDataVm)
+        {
+            try
+            {
+                doInputValidation(newPhotoDataVm);
+                PreparePhotoSaveLocation();
+            }
+            catch(Exception)
+            {
+                return ServiceResult.Fail(ErrorCodes.CouldNotSaveTheFile);
+            }
+                
+
+            var origSavePath = Path.Combine(_saveLocation, $"orginal{_uploadedFileExtension}");
+            var thumbnailSavePath = Path.Combine(_saveLocation, $"thumbnail{_uploadedFileExtension}");
+
+            try
+            {
+                await saveOriginalPhoto(origSavePath, newPhotoDataVm.file);
+                await saveThumbnail(thumbnailSavePath, newPhotoDataVm.file);
+            }
+            catch (Exception)
+            {
+                return ServiceResult.Fail(ErrorCodes.CouldNotSaveTheFile);
+            }
+
+            await SaveEntity(newPhotoDataVm);
+
+            return ServiceResult.Ok();
+        }
+
+        private async Task SaveEntity(NewPhotoDataVm newPhotoDataVm)
```

Comment:
Wydaje mi się, że ta metoda jest niepotrzebna. Chyba by było prościej przemapować mapperem to co się da i na koniec przypisać tagi.

Reply:
też o tym myślałem, ale mapperem tylko 2/5 properties bym zmapował, więc nie oszczędził by mi linii kodu, a metoda AddPhoto() była na tyle rozrośnięta że chciałem ją uprościć maksymalnie.